src/core/PalmSystemBase.cpp

```diff
@@ -41,22 +45,22 @@ QVariant PalmSystemBase::getResource(QVariant a, QVariant b)
 
 QString PalmSystemBase::country() const
 {
-    QString localcountry;
-    QString smartServiceCountry;
-    QString country;
+    std::string localcountry;
+    std::string smartServiceCountry;
 
     WebAppManager::instance()->getDeviceInfo("LocalCountry", localcountry);
     WebAppManager::instance()->getDeviceInfo("SmartServiceCountry", smartServiceCountry);
 
-    country = QString("{ \"country\": \"%1\", \"smartServiceCountry\": \"%2\" }");
-    country = country.arg(localcountry).arg(smartServiceCountry);
+    std::stringstream jss;
+    jss << "{ \"country\": \"" << localcountry
+        << "\", \"smartServiceCountry\": \"" << smartServiceCountry << "\" }";
```

Comment:
Could you take the opportunity of this change to construct JSON, then creating the string? We usually prefer to let JSON API do all the proper type checkings to construct a valid JSON content.

Reply:
This is a good point. I'll do it. But to keep the history I'll add it as a separate commit on top of the other ones. Is it ok for you?

Reply:
Done.